app/models/squad.rb

```diff
@@ -1,15 +1,13 @@
 class Squad < ApplicationRecord
-
   has_many :fleets
   belongs_to :faction
   has_many :users
 
-  def debit(value,type)
-    if type == 'credits'
-      self.update_attributes(credits: self.credits - value) unless value > self.credits
-    else
-      self.update_attributes(metals: self.metals - value) unless value > self.metals
-    end
+  def debit_credits(value)
+    update_attributes(credits: credits - value) unless value > credits
   end
 
+  def debit_metals(value)
+    update_attributes(metals: metals - value) unless value > metals
+  end
```

Comment:
👏

Reply:
Prefer `if` over `unless`, it's easier to read.
`if metals <= value`

Reply:
I prefer ifs too. fixed